flexmeasures/ui/templates/crud/asset.html

```diff
@@ -369,6 +483,436 @@ <h3>All child assets for {{ asset.name }}</h3>
 <script src="{{ url_for('flexmeasures_ui.static', filename='js/map-init.js') }}"></script>
 
+<script>
+    let units = {{available_units | tojson}};
+    const selectElement = document.getElementById('unitsSelect');
+
+    units.forEach((unit, index) => {
+        const option = document.createElement('option');
+        option.value = unit;
+        option.text = unit;
+        selectElement.appendChild(option);
+    });
+</script>
+
+<script>
+    const formModal = document.getElementById('sensorsToShowForm');
+    const apiSensorsListElement = document.getElementById("apiSensorsList");
+    const spinnerElement = document.getElementById('spinnerElement');
+    let rawJSON = "{{ asset.sensors_to_show | safe}}";
+    rawJSON = rawJSON.replace(/'/g, '"');
+    let data = JSON.parse(rawJSON);
+    let sensors = [];
+    let editingIndex; 
+    let savedGraphIndex;
+    let selectedGraphTitle;
+    const apiBasePath = window.location.origin;
+    const sensorsApiUrl = `${apiBasePath}/api/v3_0/sensors?page=1&per_page=10&asset_id={{ asset.id }}&include_public_assets=true`;
+
+    // Fetch Account Details
+    async function getAccount(accountId){
+        const cacheKey = `account_${accountId}`;
+        const cachedData = localStorage.getItem(cacheKey);
+
+        if (cachedData) {
+            return JSON.parse(cachedData);
+        }
+
+        const apiUrl = apiBasePath + "/api/v3_0/accounts/" + accountId;
+        const response = await fetch(apiUrl);
+        const account = await response.json();
+
+        localStorage.setItem(cacheKey, JSON.stringify(account));
+
+        return account;
+    }
+
+    // Fetch Asset Details
+    async function getAsset(assetId){
+        const cacheKey = `asset_${assetId}`;
+        const cachedData = localStorage.getItem(cacheKey);
+
+        if (cachedData) {
+            return JSON.parse(cachedData);
+        }
+
+        const apiUrl = apiBasePath + "/api/v3_0/assets/" + assetId;
+        const response = await fetch(apiUrl);
+        const asset = await response.json();
+
+        localStorage.setItem(cacheKey, JSON.stringify(asset));
+
+        return asset;
+    }
+
+    // Fetch Sensor Details
+    async function getSensor(id) {
+        const cacheKey = `sensor_${id}`;
+        const cachedData = localStorage.getItem(cacheKey);
+
+        if (cachedData) {
+            return JSON.parse(cachedData);
+        }
+
+        const apiUrl = apiBasePath + "/api/v3_0/sensors/" + id;
+        const response = await fetch(apiUrl);
+        const sensor = await response.json();
+        
+        localStorage.setItem(cacheKey, JSON.stringify(sensor));
+        
+        return sensor;
+    }
+
+    // highlight selected graph 
+    async function selectGraph(graphIndex) {
+        if (graphIndex !== undefined){
+            savedGraphIndex = graphIndex;
+            selectedGraphTitle = data[graphIndex]?.title;
+            apiSensorsList.innerHTML = ""; 
```

Comment:
I don't think we need to do this. The sensor search results might be useful, just the graph on the left was not the right one. If I select the right one, I have to repeat the sensor search.

Reply:
Thats true, this will be refactored. the expected behavior is to re-render the searched sensors with the currently selected graph index